src/resolutionError.ts

```diff
@@ -18,6 +18,7 @@ export enum ResolutionErrorCode {
   UnsupportedCurrency = 'UnsupportedCurrency',
   IncorrectResolverInterface = 'IncorrectResolverInterface',
   RecordNotFound = 'RecordNotFound',
+  IncorrectProvider = 'IncorrectProvider',
```

Comment:
It is better to avoid mixing resolution exceptions into the configuration excetions. You may create a `ConfigurationError` for this or just avoid a custom class for now. It is good idea to just separate expctions occur when library is used from exceptions occur when library is configured.